src/type/typeImplementation.ts

```diff
@@ -0,0 +1,89 @@
+// TODO: wrap in asyncWrapper
```

Comment:
I think this is accomplished in `src/type/index.ts` right?

```suggestion
```

Reply:
I was not sure what this TODO was referring to. Might have been accomplished a long time ago.